frontend/src/components/pages/StatisticsPage.tsx

```diff
@@ -0,0 +1,461 @@
+      bins.push(`${binMin.toFixed(0)} - ${binMax.toFixed(0)}`);
+    }
+    values.forEach(val => {
+      let idx = Math.floor((val - min) / step);
+      if (idx >= binCount) idx = binCount - 1;
+      counts[idx]++;
+    });
+    return { bins, counts };
+  }
+
+  const chartData = useMemo(() => {
+    const getAxisValue = (item: ProductRequest, attr: keyof ProductRequest | 'timestamp' | 'user_fullname') => {
+      if (attr === 'timestamp') {
+        if (Array.isArray(item.statuses) && item.statuses.length > 0) {
+          return String(item.statuses[item.statuses.length - 1].timestamp).slice(0, 10);
+        }
+        return 'нет даты';
+      }
+      if (attr === 'user_fullname') {
+        return usersList?.[String(item.user_id)] || String(item.user_id);
+      }
+      if (attr === 'category') {
+        return categoryToRussian(item.category) ?? String(item.category ?? 'null');
+      }
+      return String(item[attr] ?? 'null');
+    };
+
+    if (xAttr === yAttr) {
+      if (xAttr === 'price') {
+        const hist = buildHistogram(filteredData, 'price', 10);
+        return {
+          xLabels: hist.bins,
+          series: [{ data: hist.counts, label: 'Количество' }],
+        };
+      }
+      const counts: Record<string, number> = {};
+      filteredData.forEach(item => {
+        let val: string;
+        if (xAttr === 'category') {
+          val = categoryToRussian(item.category) ?? String(item.category ?? 'null');
+        } else if (xAttr === 'timestamp') {
+          if (Array.isArray(item.statuses) && item.statuses.length > 0) {
+            val = String(item.statuses[0].timestamp).slice(0, 10);
+          } else {
+            val = 'нет даты';
+          }
+        } else if (xAttr === 'user_fullname') {
+          val = usersList?.[String(item.user_id)] || String(item.user_id);
+        } else {
+          val = String(item[xAttr] ?? 'null');
+        }
+        counts[val] = (counts[val] ?? 0) + 1;
+      });
+      const xLabels = Object.keys(counts);
+      return {
+        xLabels,
+        series: [{
+          data: xLabels.map(x => counts[x]),
+          label: 'Количество',
+        }],
+      };
+    }
+
+    const groupMap = new Map<string, Map<string, number>>();
+
+    filteredData.forEach((item) => {
+      const xVal = getAxisValue(item, xAttr);
+      const yVal = getAxisValue(item, yAttr);
+
+      if (!groupMap.has(xVal)) groupMap.set(xVal, new Map());
+      const yMap = groupMap.get(xVal)!;
+      if (yAttr === 'price') {
+        yMap.set('price', (yMap.get('price') ?? 0) + Number(item.price ?? 0));
+      } else {
+        yMap.set(yVal, (yMap.get(yVal) ?? 0) + 1);
+      }
+    });
+
+    const xLabels = Array.from(groupMap.keys());
+    let yLabels: string[];
+    let series;
+
+    if (yAttr === 'price') {
+      yLabels = ['price'];
+      series = [{
+        data: xLabels.map((xLabel) => groupMap.get(xLabel)?.get('price') ?? 0),
+        label: 'Сумма price',
```

Comment:
Здесть стоит сделать небольшие *changes*:

```suggestion
        label: 'Сумма цен',
```